MODEL-GUIDE.md

```diff
@@ -118,3 +118,119 @@ The build script will enforce the following keyword conventions.
     ```
 
 
+## x-pattern extension
+### schema
+```yaml
+x-pattern: 
+  description: >-
+    This extension is used by the bundler to generate a unique pattern schema
+    object for properties with specific a specific type, enum (if any), default.
+  type: object
+  required: [description, format]
+  properties:
+    description: 
+      description: >-
+        Description of the parent property hosting the x-pattern extension
+      type: string  
+    format:
+      description: >-
+        Controls the shape of the generated schema object.
+      type: string
+      enum:
+      - ipv4
+      - ipv6
+      - hex
+      - integer
+      - number
+      - enum
+      - string
+      - checksum
+  length:
+    description: >-
+      The length of hex, integer and string value(s).
+      For hex and integer it is the bit length
+      For string it is the byte length
+      For a format of ipv4, ipv6 it is ignored.
+      The format will have the length appended to it: hex_4_bits, string_6_bytes
+    type: integer
+  enums: 
+    description: >-
+      If the format is enum this property is used to specify 
+      a list of enums that the pattern is constrained to
+    type: array
+    items:
+      type: string
+  default:
```

Comment:
`"1c"` ? has to be within quotes otherwise it will just be some integer for which hex format wouldn't make sense.

Reply:
I think following.
```
      x-pattern:
        format: hex
        default: 1a1c
```

NOTE, in yaml, quotes are not necessarily required for string hence it might cause confusion. All hex values need to be provided within quotes.

Reply:
in spec it would look like this
```
      x-pattern:
        format: hex
        default: 1fca
```